File: task_1/scripts/get_marker_data.py
#!/usr/bin/env python

import rospy
import roslib
import tf

from geometry_msgs.msg import PoseArray
from geometry_msgs.msg import Point

#Define a class
class Whycon_detect():

	def __init__(self):
		rospy.init_node("whycon_detection")
		self.whycon_coordinates = {}
		self.sub = rospy.Subscriber("/whycon/poses", PoseArray, self.callback)
	
	def callback(self, msg):
		for i,pos in enumerate(msg.poses):
			p = pos.position
			self.whycon_coordinates[i] = map(lambda x : round(x, 3), [p.x,p.y,p.z])
		print(self.whycon_coordinates)

if __name__=="__main__":
	marker = Whycon_detect()
	while not rospy.is_shutdown():
		rospy.spin()

File: task_1/src/get_marker_data.cpp
#include <ros/ros.h>

#include <geometry_msgs/PoseArray.h>
#include <geometry_msgs/Point.h>
#include <geometry_msgs/Pose.h>

class detect_whycon{
private:
    ros::NodeHandle n;
    ros::Subscriber s;
    geometry_msgs::Pose poses[4];

    void callback(const geometry_msgs::PoseArray::ConstPtr& msg)
    {
        for(int i = 0; i < 4; i++)
            poses[i] = msg->poses.at(i);

        std::stringstream ss;

        for(int i = 0; i < 4; i++)
        {
            std::cout << i << " : " << "{" << poses[i].position.x << "," << poses[i].position.y << "," << poses[i].position.z << "}"<<std::endl;
        }
        std::cout << std::endl;
    }

public:
    detect_whycon(){
        n = ros::NodeHandle("whycon_detection");

        s = n.subscribe("/whycon/poses", 10, &detect_whycon::callback, this);
    }
};

int main(int argc, char** argv)
{
    ros::init(argc, argv, "whycon_detection");
    detect_whycon detector = detect_whycon();
    ros::spin();
}